docs/how-to/unified_memory.rst

```diff
@@ -0,0 +1,595 @@
+                hipDeviceSynchronize();
+
+                // Prints the result.
+                std::cout << a << " + " << b << " = " << c << std::endl;
+
+                return 0;
+            }
+
+
+    .. tab-item:: malloc()
+
+        .. code-block:: cpp
+            :emphasize-lines: 12-15
+
+            #include <hip/hip_runtime.h>
+            #include <iostream>
+
+            // Addition of two values.
+            __global__ void add(int* a, int* b, int* c) {
+                *c = *a + *b;
+            }
+
+            int main() {
+                int* a, * b, * c;
+
+                // Allocate memory for a, b, and c.
+                a = (int*)malloc(sizeof(*a));
+                b = (int*)malloc(sizeof(*b));
+                c = (int*)malloc(sizeof(*c));
+
+                // Setup input values.
+                *a = 1;
+                *b = 2;
+
+                // Launch add() kernel on GPU.
+                hipLaunchKernelGGL(add, dim3(1), dim3(1), 0, 0, a, b, c);
+
+                // Wait for GPU to finish before accessing on host.
+                hipDeviceSynchronize();
+
+                // Prints the result.
+                std::cout << *a << " + " << *b << " = " << *c << std::endl;
+
+                // Cleanup allocated memory.
+                free(a);
+                free(b);
+                free(c);
+
+                return 0;
+            }
+
+
+    .. tab-item:: Explicit Memory Management
+
+        .. code-block:: cpp
+            :emphasize-lines: 17-24, 29-30
+
+            #include <hip/hip_runtime.h>
+            #include <iostream>
+
+            // Addition of two values.
+            __global__ void add(int *a, int *b, int *c) {
+                *c = *a + *b;
+            }
+
+            int main() {
+                int a, b, c;
+                int *d_a, *d_b, *d_c;
+
+                // Setup input values.
+                a = 1;
+                b = 2;
+
+                // Allocate device copies of a, b and c.
+                hipMalloc(&d_a, sizeof(*d_a));
+                hipMalloc(&d_b, sizeof(*d_b));
+                hipMalloc(&d_c, sizeof(*d_c));
+
+                // Copy input values to device.
+                hipMemcpy(d_a, &a, sizeof(*d_a), hipMemcpyHostToDevice);
+                hipMemcpy(d_b, &b, sizeof(*d_b), hipMemcpyHostToDevice);
+
+                // Launch add() kernel on GPU.
+                hipLaunchKernelGGL(add, dim3(1), dim3(1), 0, 0, d_a, d_b, d_c);
+
+                // Copy the result back to the host.
+                hipMemcpy(&c, d_c, sizeof(*d_c), hipMemcpyDeviceToHost);
+
+                // Cleanup allocated memory.
+                hipFree(d_a);
+                hipFree(d_b);
+                hipFree(d_c);
+
+                // Prints the result.
+                std::cout << a << " + " << b << " = " << c << std::endl;
+
+                return 0;
+            }
+
+
+Compiler Hints for the Better Performance
```

Comment:
Could you help establish context with UMM? Do we mean C++ compiler hints?

Reply:
These are UMM compiler hints. I updated the header.